spec/models/solidus_avatax_certified/request/get_tax_spec.rb

```diff
@@ -11,16 +11,42 @@
   end
 
   describe '#generate' do
-    it 'creates a hash' do
-      expect(subject.generate).to be_kind_of Hash
+    subject(:generate) { service.generate }
+
+    shared_examples 'build tax request' do
+      it 'creates a hash' do
+        expect(generate).to be_kind_of Hash
+      end
+
+      it 'Commit has value of false' do
+        expect(generate[:createTransactionModel][:commit]).to be false
+      end
+
+      it 'has ReferenceCode from base_tax_hash' do
+        expect(generate[:createTransactionModel][:referenceCode]).to eq(order.number)
+      end
     end
 
-    it 'Commit has value of false' do
-      expect(subject.generate[:createTransactionModel][:commit]).to be false
+    context 'when store uses default config' do
+      before do
+        Spree::AvataxConfiguration.company_code = 'DEFAULT_COMPANY_CODE'
+      end
+
+      it_behaves_like 'build tax request'
+
+      it 'uses default company code' do
+        expect(generate[:companyCode]).to eq('DEFAULT_COMPANY_CODE')
+      end
     end
 
-    it 'has ReferenceCode from base_tax_hash' do
-      expect(subject.generate[:createTransactionModel][:referenceCode]).to eq(order.number)
+    context 'when store has its own config' do
+      before do
+        order.update(avatax_config:  { company_code: 'STORE_COMPANY_CODE' })
```

Comment:
I'm not 100% familiar, but shouldn't it be `order.store.update`?

Reply:
yeah - fixed